src/DynamoCore/Core/NodeModelAssemblyLoader.cs

```diff
@@ -199,6 +199,52 @@ internal static bool ContainsNodeModelSubType(Assembly assem)
             return assem.GetTypes().Any(IsNodeSubType);
         }
 
+        internal static bool ContainsNodeModelSubTypeReflectionLoaded(Assembly assem, Type nodeModelType = null)
+        {
+            if(nodeModelType is null)
+                return assem.GetTypes().Any(x => IsNodeSubType(x));
+            
+            return assem.GetTypes().Any(x => IsNodeSubTypeReflectionLoaded(x, nodeModelType));
+        }
+
+        /// <summary>
+        /// Similar to IsNodeSubType this will Determine if a Type is a node.
+        /// This implementation is meant to be used when scanning types in ReflectionContext.
+        /// </summary>
+        /// <param name="t">Type to check</param>
+        /// <param name="nodeModelType">Type of NodeModel, this is needed to be sure we are comparing the same NodeModel type reference</param>
+        /// <returns></returns>
+        internal static bool IsNodeSubTypeReflectionLoaded(Type t, Type nodeModelType)
+        {
+            bool isNodeSubType = false;
+            try
+            {
+                isNodeSubType = !t.IsAbstract &&
+                    t.IsSubclassOf(nodeModelType) &&
+                    t.GetConstructor(Type.EmptyTypes) != null;
+            }
+            catch
+            {
+                // If for any reason the above fails we try to determine if the type is a node by checking for NodeModel attributes.
+                // Above can potentially fail if there is a reference to a type which is not included in the path resolver.
+                var customAttributes = CustomAttributeData.GetCustomAttributes(t).ToList();
+                var customAttributeTypes = customAttributes.Select(x => x.AttributeType).ToList();
+
+                if (customAttributes is null || customAttributes.Count == 0)
+                    return false;
+
+                isNodeSubType = customAttributes.
```

Comment:
thats an interesting workaround - did you see this somewhere else? Is this guaranteed? Can you only apply these attributes to NodeModel types or really to any class?

Reply:
I think I did, but going over it again I probably didn't...
Needed this as sometimes i had issues with getting the IsSubclassOf but should be fine now as we should have all needed references loaded.